skbio/diversity/alpha/_base.py

```diff
@@ -316,10 +426,17 @@ def faith_pd(counts, otu_ids, tree, validate=True):
     if validate:
         counts = _validate_counts_vector(counts)
         _validate_otu_ids_and_tree(counts, otu_ids, tree)
-    observed_otus = {o: c for o, c in zip(otu_ids, counts) if c >= 1}
-    observed_nodes = tree.observed_node_counts(observed_otus)
-    result = sum(o.length for o in observed_nodes if o.length is not None)
-    return result
+
+    counts = np.asarray(counts)
+    otu_ids = np.asarray(otu_ids)
```

Comment:
Move this to ``_vectorize_counts_and_tree``, since that's what cares about it, and so we don't have to always make sure that we're passing a numpy array.